feat: make rigel generated dockerfile pretty(#15)

miguelsantos14 opened the pull request.

MisterOwlPT reviewed (changes requested):
Hi @miguelsantos14!
You are again on the right path 👍🏻 

**Some remarks on your changes:**

- The Jinja template files are _not_ to be placed at the root directory of the repository. They only concern the Dockerfile generation plugin and as such, should be placed within the plugin directory, in the sub directory assigned for such files. If applicable, don't forget to update paths after moving the mentioned files to their new location.

- You made some changes to the original Rigelfile template. This file is used as is by the command `rigel init`. It is meant to be a very basic and ready-to-use Rigelfile. Therefore it should not be changed, unless to be updated to the latest Rigelfile specification and should _not_ contain anything specific to your use case or tests. Please undo these changes and use a separate Rigelfile for your tests.

miguelsantos14 commented:
Hi @MisterOwlPT 
I commited the changes with the new files, as you told me to do.
I expect everything to be fine!

MisterOwlPT commented on rigel/plugins/core/dockerfile/assets/Dockerfile.j2:
```diff
@@ -1,208 +1,63 @@
 
-# Copy dependencies from intermediate stage.
-COPY --from=intermediate /ros_workspace/src /home/{{ configuration.username }}/ros_workspace/src/
+COPY rigel/plugins/core/dockerfile/assets/python_dependencies.dockerfile.j2 /home/rigeluser/ros_workspace/
```
Why are you copying the template files into the final container image?
**The template files are used to generate the Dockerfile. They should not be imported into the container image.**

MisterOwlPT commented on rigel/plugins/core/dockerfile/assets/Dockerfile.j2:
```diff
@@ -1,208 +1,63 @@
-RUN sh -c 'sudo chmod +x /home/{{ configuration.username }}/robot-entrypoint.sh'
-RUN sh -c 'sudo chown {{ configuration.username }}:{{ configuration.username }} /home/{{ configuration.username }}/robot-entrypoint.sh'
+# Set the entrypoint to launch the ROS package
+ENTRYPOINT ["/bin/bash", "-c", "source /opt/ros/noetic/setup.bash && source /home/rigeluser/ros_workspace/devel/setup.bash && roslaunch your_package your_launch_file.launch"]
```
Why are you forcing `/opt/ros/noetic` here?
What if the user is targeting other ROS distributions?
Was this on the original template?

MisterOwlPT reviewed (changes requested):
Hi @miguelsantos14,

This is still work to do before your changes can be merged:
- I noticed that you copied the Jinja template files into the proper plugin folder. However, **copies of the template files are still on the root of the repository**. Delete them;
- **The template Rigelfile still contains your changes.** As mentioned before, revert these changes to the original template;
-

miguelsantos14 commented:
Hello @MisterOwlPT!
All the changes requested are done, it was really unnecessary the way I was forcing  /opt/ros/noetic, and of course it isn't necessary to copy the templates files into the final container image.
I also reverted the Rigelfile and removed the requested files.
I think that now it's ready to merge!

MisterOwlPT commented:
Hello @miguelsantos14

As requested you have removed the Jinja template files from the root directory 👍🏻 

However problems _persist_ with your changes. I `git pull`ed several times to ensure I had indeed the most recent changes:
- The original Rigelfile template is still configured to your test case. **Revert these changes.**
- You are still forcing `noetic` in places like [here](https://github.com/rigel-ros/rigel/blob/feat/make-rigel-generated-dockerfile-pretty/rigel/plugins/core/dockerfile/assets/Dockerfile.j2#L47) and [here](https://github.com/rigel-ros/rigel/blob/feat/make-rigel-generated-dockerfile-pretty/rigel/plugins/core/dockerfile/assets/Dockerfile.j2#L15) (`noetic` uses `python3` while older ROS distros do not).

I think there must have been a misunderstanding somewhere... it looks like you replaced the original template with the Dockerfile from your test case. Is this correct?

For instance, where are you importing the smaller Jinja templates? You are [rendering the Dockerfile using the `Dockerfile.j2` template](https://github.com/rigel-ros/rigel/blob/feat/make-rigel-generated-dockerfile-pretty/rigel/plugins/core/dockerfile/plugin.py#L71) and it does not reference any other template files 🤔

The idea here is to replace the big monolithic Dockerfile template with multiple, more granular templates that are easier to manage (alternative generation strategies are welcomed). **These smaller templates _must_  be combined (i.e., properly parameterized and rendered in order) as to generate the _exact same Dockerfile_ as before**,  without any changes (except the unnecessary newlines and weird spacing here and there).

Let me know is this makes sense to you.
Feel free to ask any question if something is not clear 🙂

miguelsantos14 commented:
Hello @MisterOwlPT 
I apologize for all the mistakes and the confusion around this issue. 
Now, I think that the goal is being reached. I am generating the original big Dockerfile, using small templates.
Please tell me if anything is wrong or if I should have more content in the generated Dockerfile.

miguelsantos14 commented:
Hello @MisterOwlPT 
Changes done. Now, the Dockerfile that is being generated is exactly the same that was when we had only the big Dockerfile.j2
I think that this task is finally done.

MisterOwlPT reviewed (approved):
Hi @miguelsantos14 
Nice work! 🙌🏻 
Now we can merge your changes.
